pkg/cli/cmds/states.go

```diff
@@ -81,18 +101,111 @@ func stateList(c *cli.Context) error {
 	return nil
 }
 
-func createExecution(c *cli.Context) error {
+func stateShow(c *cli.Context) error {
+	kubeConfig := c.GlobalString("kubeconfig")
+	namespace := c.GlobalString("namespace")
+
+	if len(c.Args()) < 1 {
+		logrus.Fatal("No state name passed.")
+	}
+
+	name := c.Args()[0]
+
+	state, err := getState(namespace, kubeConfig, name)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("State: %s\n", name)
+	fmt.Printf("Auto Confirm: %t\n", state.Spec.AutoConfirm)
+	fmt.Printf("Destroy on Deleted: %t\n", state.Spec.AutoConfirm)
+	fmt.Println("")
+	if len(state.Spec.Variables.EnvConfigName) > 0 {
+		for _, value := range state.Spec.Variables.EnvConfigName {
+			config, err := getConfig(namespace, kubeConfig, value)
+			if err != nil {
+				if k8sError.IsNotFound(err) {
+					fmt.Printf("missing info: secret %s\n", value)
+					continue
+				}
+			}
+
+			for key, value := range config.Data {
+				fmt.Printf("%s: %s\n", key, value)
+			}
+		}
+	}
+
+	fmt.Println("")
+	if len(state.Spec.Variables.SecretNames) > 0 {
+		for _, secretName := range state.Spec.Variables.SecretNames {
+			secret, err := getSecret(namespace, kubeConfig, secretName)
+			if err != nil {
+				if k8sError.IsNotFound(err) {
+					fmt.Printf("missing info: secret %s\n", secretName)
+					continue
+				}
+			}
+
+			for key, value := range secret.Data {
+				fmt.Printf("%s: %s\n", key, value)
+			}
+		}
+	}
+
+	return nil
+}
+
+func stateUnlock(c *cli.Context) error {
+	kubeConfig := c.GlobalString("kubeconfig")
+	namespace := c.GlobalString("namespace")
+	controllers, err := getControllers(kubeConfig, namespace)
+	if err != nil {
+		return err
+	}
+
+	args := c.Args()
+	if len(args) != 1 {
+		return cli.NewExitError("No state name passed", 1)
+	}
+	name := c.Args().First()
+
+	selector := fmt.Sprintf("%s=true,%s=%s", terraState, terraKey, name)
+	secrets, err := controllers.secrets.List(namespace, metav1.ListOptions{
+		LabelSelector: selector,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	for _, secret := range secrets.Items {
+		if secret.Data["lockInfo"] != nil {
+			copy := secret.DeepCopy()
+			copy.Data["lockInfo"] = nil
+			_, err := controllers.secrets.Update(copy)
+			if err != nil {
+				return err
+			}
+			logrus.Infof("Unlocking %s", name)
+		}
+	}
+
+	return nil
+}
+
+func createState(c *cli.Context) error {
 	kubeConfig := c.GlobalString("kubeconfig")
 	namespace := c.GlobalString("namespace")
 
 	if len(c.Args()) != 2 {
 		return InvalidArgs{}
```

Comment:
I don't know why this can't be an Errof() return? Why do we need to use a empty struct?